llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -7282,7 +7282,7 @@ bool SPIRVToLLVM::checkContains64BitType(SPIRVType *bt) {
 
 bool SPIRVToLLVM::transShaderDecoration(SPIRVValue *bv, Value *v) {
   auto gv = dyn_cast<GlobalVariable>(v);
-  if (gv) {
+  if (gv && bv->getOpCode() == OpVariable) {
```

Comment:
```suggestion
  // Some SPIR-V instructions (e.g. OpAccessChain) can become no-ops in LLVM IR, so we must
  // explicitly check both the SPIR-V opcode and the LLVM type.
  if (gv && bv->getOpCode() == OpVariable) {
```

Reply:
Makes sense, but easy to miss in the future so please add a comment e.g. as suggested.